packages/bug-detectors/internal/prototype-pollution.ts

```diff
@@ -0,0 +1,565 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AssignmentExpression, Identifier, Node } from "@babel/types";
+import { NodePath, PluginTarget, types } from "@babel/core";
+import { reportFinding } from "../findings";
+import {
+	registerAfterEachCallback,
+	addDictionary,
+	registerInstrumentationPlugin,
+} from "@jazzer.js/hooking";
+
+import { bugDetectorConfigurations } from "../configuration";
+
+// Allow the user to configure this bug detector in the custom-hooks file (if any).
+class PrototypePollutionConfig {
+	private excludedExactMatches: string[] = [];
+	private _instrument = false;
+
+	/**
+	 * Excludes one specific value of the `__proto__` property from being reported as a Prototype Pollution finding.
+	 * This is only relevant when instrumenting assignment expressions and variable declarations.
+	 * @param protoValue - stringified value of the `__proto__` for which no finding should be reported.
+	 */
+	addExcludedExactMatch(protoValue: string): PrototypePollutionConfig {
+		this.excludedExactMatches.push(protoValue);
+		return this;
+	}
+
+	/**
+	 * Enables instrumentation of assignment expressions and variable declarations.
+	 * This is a costly operation that might find non-global Prototype Pollution.
+	 * However, it also might result in false positives. Use `addExcludedExactMatch` to exclude specific values from
```

Comment:
I would really like to see an example of a possible false positive here.

Reply:
I added some examples in the comments.
If you're curious: variable `freeProcess.__proto__`  in the lodash library has `'{"_eventsCount":0}'` while the library is loaded.
Or protobufjs after some fuzzing:
```
Prototype Pollution: service.__proto__ value is {"methods":{}}
Prototype Pollution: type.__proto__ value is {"fields":{}}
```